dart/test/sentry_log_batcher_test.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:test/test.dart';
+import 'package:sentry/src/sentry_log_batcher.dart';
+import 'package:sentry/src/sentry_options.dart';
+import 'package:sentry/src/protocol/sentry_log.dart';
+import 'package:sentry/src/protocol/sentry_log_level.dart';
+
+import 'mocks/mock_transport.dart';
+
+void main() {
+  late Fixture fixture;
+
+  setUp(() {
+    fixture = Fixture();
+  });
+
+  test('added logs are flushed after timeout', () async {
+    final flushTimeout = Duration(milliseconds: 1);
+
+    final batcher = fixture.getSut(flushTimeout: flushTimeout);
+
+    final log = SentryLog(
+      timestamp: DateTime.now(),
+      level: SentryLogLevel.info,
+      body: 'test',
+      attributes: {},
+    );
+    final log2 = SentryLog(
+      timestamp: DateTime.now(),
+      level: SentryLogLevel.info,
+      body: 'test2',
+      attributes: {},
+    );
+
+    batcher.addLog(log);
+    batcher.addLog(log2);
+
+    expect(fixture.mockTransport.envelopes.length, 0);
+
+    await Future.delayed(flushTimeout);
+
+    expect(fixture.mockTransport.envelopes.length, 1);
+
+    final envelopePayloadJson = (fixture.mockTransport).logs.first;
+
+    expect(envelopePayloadJson, isNotNull);
+    expect(envelopePayloadJson['items'].length, 2);
+    expect(envelopePayloadJson['items'].first['body'], log.body);
+    expect(envelopePayloadJson['items'].last['body'], log2.body);
+  });
+
+  test('max logs are flushed immediately', () async {
+    final batcher = fixture.getSut(maxBufferSize: 10);
+
+    final log = SentryLog(
+      timestamp: DateTime.now(),
+      level: SentryLogLevel.info,
+      body: 'test',
+      attributes: {},
+    );
+
+    for (var i = 0; i < 10; i++) {
+      await batcher.addLog(log);
+    }
+
+    expect(fixture.mockTransport.envelopes.length, 1);
+    final envelopePayloadJson = (fixture.mockTransport).logs.first;
+
+    expect(envelopePayloadJson, isNotNull);
+    expect(envelopePayloadJson['items'].length, 10);
+  });
```

Comment:
can you also add a test that when the buffer is flushed due to max size, that the rest of the logs end up in the buffer

e.g

Add 105 logs
 - 100 logs are flushed
 - 5 logs are processed into the buffer so we don't lose them

Reply:
This is already handled. Added a test to verify.